pkg/ipcache/ipcache.go

```diff
@@ -189,6 +191,21 @@ const (
 	Delete CacheModification = "Delete"
 )
 
+func keyToIP(key string) net.IP {
+	fullPath := path.Join(path.Join(IPIdentitiesPath, AddressSpace))
+	if !strings.HasPrefix(key, fullPath) {
+		log.WithFields(logrus.Fields{"key": key, "prefix": IPIdentitiesPath}).Warning("Found invalid key outside of prefix")
+		return nil
+	}
+
+	suffix := strings.TrimPrefix(key, fullPath)
```

Comment:
Add the `/` to the `requiredPrefix` so gets trimmed away? I don't see a case where we have a valid suffix without a `/`.